src/ripple/protocol/Indexes.h

```diff
@@ -184,13 +184,13 @@ check(uint256 const& key) noexcept
 
 /** A DepositPreauth */
 /** @{ */
-Keylet
+DPAuthKeylet
 depositPreauth(AccountID const& owner, AccountID const& preauthorized) noexcept;
 
-inline Keylet
+inline DPAuthKeylet
 depositPreauth(uint256 const& key) noexcept
 {
-    return {ltDEPOSIT_PREAUTH, key};
+    return {key};
```

Comment:
ok. I have a tangential question, what are the tradeoffs with `return DPAuthKeyley{key}` ?

I am aware the there could multiple ways to initialize objects, but I am not clear about the differences.
Since this is an aggregate type, I believe using the `{}` syntax should be okay, but IDK the performance costs.

Reply:
The runtime costs of `()` vs `{}` are identical.  The difference is in how permissive the syntaxes are.

This is really guidance that I got from Howard.  If I'm remembering correctly, there are situations where braced initialization will consider using an `explicit` constructor.  So if you just casually use `{}` to initialize stuff, then you may be invoking an `explicit` constructor without realizing it.  Howard said that using `=` or `()` for initialization does not have that problem.

For me, at least, C++ is a complicated enough language that I don't want to think, "Hmmm, what kind of initialization should I use here?" every time I initialize a value.  So as a heuristic I tend not to use braced initialization unless I have a specific reason.

Note that Howard's guidance disagrees with guidance that Nico Jossutis (at least used to) give.  Nico used to like defaulting to braced initialization (without the `=`).

As I'm looking around now I think you'll see that the safest default is `= {<value(s)>}`: https://www.josuttis.com/cpp/c++initialization.pdf  But I'm not convinced that we could convince rippled developers to type that many characters every time they initialize initialize something.

That's my current take.  But I suggest you talk to Howard about it.  He has opinions and did the tests to back them up.  Let me know what he says.

Reply:
will do, thanks for sharing your thoughts.
Yes, I saw Nicolai Josutis's talk on the NIghtmare of Initialization. It was a fascinating talk :)